RecoEgamma/EgammaMCTools/plugins/PFClusterMatchedToPhotonsSelector.cc

```diff
@@ -0,0 +1,114 @@
+// -*- C++ -*-
+//
+// Package:    CommonTools/RecoAlgos
+// Class:      PFClusterMatchedToPhotonsSelector
+// 
+/**\class PFClusterMatchedToPhotonsSelector PFClusterMatchedToPhotonsSelector.cc CommonTools/RecoAlgos/plugins/PFClusterMatchedToPhotonsSelector.cc
+
+ Description: Matches ECAL PF clusters to photons that do not convert
+
+ Implementation:
+
+*/
+//
+// Original Author:  RCLSA
+//         Created:  Wed, 22 Mar 2017 18:01:40 GMT
+//
+//
+
+
+// system include files
+#include <memory>
+#include <iostream>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFClusterFwd.h"
+#include "DataFormats/HepMCCandidate/interface/GenParticle.h"
+#include "DataFormats/HepMCCandidate/interface/GenParticleFwd.h"
+#include "SimDataFormats/TrackingAnalysis/interface/TrackingParticle.h"
+#include "SimDataFormats/TrackingAnalysis/interface/TrackingParticleFwd.h"
+
+
+class PFClusterMatchedToPhotonsSelector : public edm::EDProducer {
```

Comment:
Is there any reason for it being a legacy module?
Shouldn't it better inherit from public edm::stream::EDProducer<> ?